drivers/flash/flash_npcx_fiu_nor.c

```diff
@@ -284,20 +287,26 @@ static int flash_npcx_nor_erase(const struct device *dev, off_t addr,
 		/* Send chip erase command */
 		flash_npcx_uma_cmd_only(dev, SPI_NOR_CMD_CE);
 		return flash_npcx_nor_wait_until_ready(dev);
-	} else if (config->min_erase_size == KB(4)) {
-		opcode = SPI_NOR_CMD_SE;
-	} else if (config->min_erase_size == KB(64)) {
-		opcode = SPI_NOR_CMD_BE;
-	} else {
-		return -EINVAL;
 	}
 
 	while (size > 0) {
 		flash_npcx_uma_cmd_only(dev, SPI_NOR_CMD_WREN);
 		/* Send page/block erase command with addr */
-		flash_npcx_uma_cmd_by_addr(dev, opcode, addr);
-		addr += config->min_erase_size;
-		size -= config->min_erase_size;
+		if ((size >= BLOCK_64K_SIZE) && offset_64k_aligned) {
+			flash_npcx_uma_cmd_by_addr(dev, SPI_NOR_CMD_BE, addr);
+			addr += BLOCK_64K_SIZE;
+			size -= BLOCK_64K_SIZE;
+		} else if ((size >= BLOCK_32K_SIZE) && offset_32k_aligned) {
+			flash_npcx_uma_cmd_by_addr(dev, SPI_NOR_CMD_BE_32K, addr);
+			addr += BLOCK_32K_SIZE;
+			size -= BLOCK_32K_SIZE;
```

Comment:
@MulinChao could you check if this is supported here?

Reply:
32KB block erase support depends on spi nor flash itself. So far, all internal flashes (winbond spi flashes) support both 64KB and 32KB block erasing. Considering @de-nordic 's comment, the other spi nor flashes should be supported since the users might choose them via the external path. 
How about:
1. Add `max-erase-size` and `min-erase-size` back to nuvoton,npcx-fiu-nor.yaml.
```
  max-erase-size:
    type: int
    default: 0x10000
    description: Maximum erase size of spi nor flash.
    enum:
      - 0x10000 # 64KB (64KB Block Erase)
      - 0x8000  # 32KB (32KB Block Erase)
      - 0x1000  # 4KB (Sector Erase)
	  
  min-erase-size:
    type: int
    default: 0x10000
    description: Minimum erase size of spi nor flash.
    enum:
      - 0x1000  # 4KB (Sector Erase)
``` 
2.  size must be aligned with  `min-erase-size`
3.  if `max-erase-size` > `min-erase-size` and size > `max-erase-size`, then use the corresponding command of `max-erase-size` to erase flash for better performance. I don't think we need to proceed with two kinds of block erase at the same time. If nor flash supports 64KB block erasing, use it directly. Or use 32KB block erasing instead.

Reply:
Are 4k and 64k always present on SPI flashes? Wondering if we could get away with that and saves the driver from the option entirely.

Reply:
>Are 4k and 64k always present on SPI flashes?

Per my experience, most SPI nor flashes should support SE (4kb) and BE (64kb) commands. The followings are the datasheets I have:

***Support SE, BE, and BE32K commands:***
winbond:
w25qxxx series
w25rxxx series
mxic:
MX25Lxxx series
MX77Lxxx series
GigaDevice:
gd25qxxx series
Cypress:
S25FLxxx series

***Support SE and BE commands only:***
Micron:
N25Qxxxx series

Reply:
> Are 4k and 64k always present on SPI flashes? Wondering if we could get away with that and saves the driver from the option entirely. This is complex enough without adding other variations and conditions, maybe we can just have the common case and that's it for now.

The problem is that what we should have is actually flash drivers which would provide basic information for onboard SPI Flash controllers, that could then use it to issue proper commands. For example
```
&qspi {
        status = "okay";
        pinctrl-0 = <&qspi_default>;
        pinctrl-1 = <&qspi_sleep>;
        pinctrl-names = "default", "sleep";
        mx25r64: mx25r6435f@0 {
                compatible = "nordic,qspi-nor";
                reg = <0>;
                /* MX25R64 supports only pp and pp4io */
                writeoc = "pp4io";
                /* MX25R64 supports all readoc options */
                readoc = "read4io";
                sck-frequency = <8000000>;
                jedec-id = [c2 28 17];
                sfdp-bfp = [
                        e5 20 f1 ff  ff ff ff 03  44 eb 08 6b  08 3b 04 bb
                        ee ff ff ff  ff ff 00 ff  ff ff 00 ff  0c 20 0f 52
                        10 d8 00 ff  23 72 f5 00  82 ed 04 cc  44 83 68 44
                        30 b0 30 b0  f7 c4 d5 5c  00 be 29 ff  f0 d0 ff ff
                ];
                size = <67108864>;
                has-dpd;
                t-enter-dpd = <10000>;
                t-exit-dpd = <35000>;
        };
};
```
this, taken from nrf52840dk has `compatible = "nordic,qspi-nor";` because it is used by the qspi-nor controller, while controller should  have actually Flash device hanging from it with `compatible = "macronix,mx25rxxx"` and controller should ask the device what are its capabilities. But that would require some major rework.

Reply:
I'd suggest dropping 32k support for now (leave the `SPI_NOR_IS_32K_ALIGNED` macro, that may come in handy)  and then we can rework the extra details later if needed by an application.